.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+          LOG_GROUP_NAME: otel-sdk-performance-tests
+          # Also uses:
+          # AWS_ACCESS_KEY_ID
+          # AWS_SECRET_ACCESS_KEY
+          # AWS_SESSION_TOKEN
+          # APP_PATH
+          # TARGET_SHA
+          # LISTEN_ADDRESS_PORT
+          # LOGS_NAMESPACE
+          # APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE
+          # APP_PROCESS_EXECUTABLE_NAME
+          # HOSTMETRICS_INTERVAL_SECS
+          # TEST_DURATION_MINUTES
+          # NUM_OF_CPUS
+          # CPU_LOAD_THRESHOLD
+          # TOTAL_MEMORY_THRESHOLD
+          # AWS_DEFAULT_REGION
+          # MATRIX_COMMIT_COMBO
+          # GITHUB_RUN_ID
+        run: |-
+          docker-compose up --build;
+          RUN_TESTS_EXIT_CODE=$(
+            docker inspect $(
+              docker ps --quiet --all --filter "name=docker-performance-tests_alarms-poller"
+            ) --format="{{.State.ExitCode}}"
+          );
+          echo "RUN_TESTS_EXIT_CODE=$RUN_TESTS_EXIT_CODE" | tee --append $GITHUB_ENV;
+          exit $RUN_TESTS_EXIT_CODE;
+      - name: Fail early if Soak Tests failed to start
+        if: ${{ env.RUN_TESTS_EXIT_CODE == '' || env.RUN_TESTS_EXIT_CODE == 1 }}
+        run: exit 1;
+
+    # MARK: - Report on Performance Test Results
+
+      - name: Install script dependencies
+        run: pip install boto3
+      - name: Get a snapshot of metrics and commit them to the repository
+        run: |
+          python3 .github/scripts/performance-tests/produce_metric_widget_images.py \
+            --logs-namespace ${{ env.LOGS_NAMESPACE }} \
+            --metrics-period ${{ env.HOSTMETRICS_INTERVAL_SECS }} \
+            --num-of-cpus ${{ env.NUM_OF_CPUS }} \
+            --app-process-command-line-dimension-value "${{ env.APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE }}" \
+            --target-sha ${{ env.TARGET_SHA }} \
+            --github-run-id ${GITHUB_RUN_ID} \
+            --test-duration-minutes ${{ env.TEST_DURATION_MINUTES }} \
+            --cpu-load-threshold ${{ env.CPU_LOAD_THRESHOLD }} \
+            --total-memory-threshold ${{ env.TOTAL_MEMORY_THRESHOLD }} \
+            --app-platform ${{ matrix.app-platform }} \
+            --instrumentation-type ${{ matrix.instrumentation-type }} \
+            --max-benchmarks-to-keep ${{ env.MAX_BENCHMARKS_TO_KEEP }} \
+            --github-repository ${GITHUB_REPOSITORY}
+          echo "::warning::Checkout Snapshots at this link: https://github.com/${GITHUB_REPOSITORY}/blob/gh-pages/soak-tests/snapshots/commits/${{ env.TARGET_SHA }}/runs/${GITHUB_RUN_ID}/${{ matrix.app-platform }}";
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com";
+          git config user.name "GitHub Actions";
+          git fetch;
+          git checkout gh-pages;
+          git add soak-tests/snapshots/commits;
+          git commit -m "Soak Test Snapshots from ${{ env.TARGET_SHA }} - ${GITHUB_RUN_ID}";
+          git push;
+          git checkout main;
```

Comment:
Should this be checking out main or the previous working branch? I think you can use `git checkout -` to go back to the previous branch before the `gh-pages` checkout.

Reply:
You're right that seems to work for me! But I like the guarantee of knowing we'll be on `main` at the end of this step. Sorry if I missed something else, let me know if there's another benefit of having it be `git checkout -`?